env0/resource_environment.go

```diff
@@ -277,43 +278,97 @@ func setEnvironmentSchema(d *schema.ResourceData, environment client.Environment
 	return nil
 }
 
+func createVariable(configurationVariable *client.ConfigurationVariable) interface{} {
+	variable := make(map[string]interface{})
+
+	variable["name"] = configurationVariable.Name
+	variable["value"] = configurationVariable.Value
+
+	if configurationVariable.Type == nil || *configurationVariable.Type == 0 {
+		variable["type"] = "environment"
+	} else {
+		variable["type"] = "terraform"
+	}
+
+	if configurationVariable.Description != "" {
+		variable["description"] = configurationVariable.Description
+	}
+
+	if configurationVariable.Regex != "" {
+		variable["regex"] = configurationVariable.Regex
+	}
+
+	if configurationVariable.IsSensitive != nil {
+		variable["is_sensitive"] = configurationVariable.IsSensitive
+	}
+
+	if configurationVariable.IsReadOnly != nil {
+		variable["is_read_only"] = configurationVariable.IsReadOnly
+	}
+
+	if configurationVariable.IsRequired != nil {
+		variable["is_required"] = configurationVariable.IsRequired
+	}
+
+	if configurationVariable.Schema != nil {
+		variable["schema_type"] = configurationVariable.Schema.Type
+		variable["schema_enum"] = configurationVariable.Schema.Enum
+		variable["schema_format"] = configurationVariable.Schema.Format
+	}
+
+	return variable
+}
+
 func setEnvironmentConfigurationSchema(d *schema.ResourceData, configurationVariables []client.ConfigurationVariable) {
-	var variables []interface{}
+	ivariables, ok := d.GetOk("configuration")
```

Comment:
date_environment does not have the configuration field. Without GetOk, this will crash.